etcdserver/server.go

```diff
@@ -747,7 +747,8 @@ func (s *EtcdServer) run() {
 					}
 					lid := lease.ID
 					s.goAttach(func() {
-						s.LeaseRevoke(s.ctx, &pb.LeaseRevokeRequest{ID: int64(lid)})
+						ctx := context.WithValue(s.ctx, "LeaseRevokeExpire", struct{}{})
```

Comment:
Checking for a context value each time in AuthInfoFromCtx just for lease expiration seems wasteful. How about something like:
```go
ctx := s.authStore.WithRoot(s.ctx)
```